osquery/tables/system/windows/drivers.cpp

```diff
@@ -173,79 +146,140 @@ Status getDeviceProperty(const device_infoset_t& infoset,
   return Status();
 }
 
-static inline std::string getDriverImagePath(const std::string& service_key) {
+std::string getDriverImagePath(const std::string& service_key) {
   SQL sql("SELECT data FROM registry WHERE path = '" + service_key +
           "\\ImagePath'");
-  if (sql.rows().size() == 1) {
-    auto path = sql.rows().at(0).at("data");
-    if (!path.empty()) {
-      CHAR systemRoot[MAX_PATH] = {0};
-      GetSystemDirectory(systemRoot, MAX_PATH);
-      return systemRoot +
-             boost::regex_replace(path, boost::regex("^.*[Ss]ystem32"), "");
-    }
+
+  if (sql.rows().empty() || sql.rows().at(0).count("data") == 0) {
+    return "";
   }
-  return "";
+
+  auto path = sql.rows().at(0).at("data");
+  if (path.empty()) {
+    return "";
+  }
+
+  // Unify the image path as systemRoot can contain systemroot and/or system32
+  std::transform(path.begin(), path.end(), path.begin(), ::tolower);
+
+  char systemRoot[MAX_PATH] = {0};
+  GetSystemDirectory(systemRoot, MAX_PATH);
+
+  return systemRoot +
+         boost::regex_replace(path, boost::regex("^.*?system32"), "");
 }
 
 QueryData genDrivers(QueryContext& context) {
   QueryData results;
-
-  auto devInfoset = setupDevInfoSet();
+  auto devInfoset = setupDevInfoSet(DIGCF_ALLCLASSES | DIGCF_PRESENT);
   if (devInfoset == nullptr) {
     win32LogWARNING("Error getting device handle");
     return results;
   }
 
+  std::map<std::string, Row> apiDevices;
```

Comment:
`unordered_map` will give you more performance on lookups at the cost of memory. Might be worth it depending on how many accesses/insertions you do.